### ISSUE TITLE
Undefined runtime exception with Dialog polyfill

### ISSUE THREAD

**Issue by FabienHenon** [debois/elm-mdl#284]
I'm trying to use the `Dialog` component and I included the polyfill as mentioned in the documentation: 

I added these lines in my `index.js` : 

```js
require('dialog-polyfill/dialog-polyfill.css');

window.dialogPolyfill = require('dialog-polyfill/dialog-polyfill.js');

var root = document.getElementById('root');
var obj = {/*obj*/};
var app = Elm.Main.embed(root, obj);
```

Here is how I use the dialog component:

```elm
dialog : Material.Model -> Model -> Html Msg
dialog mdl model =
    Dialog.view
        []
        [ Dialog.title [] []
        , Dialog.content [] [ viewForm mdl model ]
        , Dialog.actions [] []
        ]


viewForm : Material.Model -> Model -> Html Msg
viewForm mdl model =
    Textfield.render Mdl
        [ 0 ]
        mdl
        [ Textfield.label "Input"
        , Textfield.floatingLabel
        , Textfield.text_
        , Material.Options.onInput (OnChange Name)
        , Textfield.value model.name
        ]
        []
```

The `dialog` function is called in a `div` directly in the root `div`.

In Chrome and Opera this is working fine (I think they are not using the polyfill), but in FF and Safari I got an exception (I didn't test on IE/Edge). Here is the exception I get:

```
TypeError: undefined is not an object (evaluating 'domNode.childNodes')
```

When I look at the stack trace I can see that the error is coming from the `addDomNodesHelp` core Elm function. And when I look at the variables I can see that `domNode` is not defined because of this code in `addDomNodesHelp` function:

```js
case 'node':
			var vChildren = vNode.children;
			var childNodes = domNode.childNodes;
			for (var j = 0; j < vChildren.length; j++)
			{
				low++;
				var vChild = vChildren[j];
				var nextLow = low + (vChild.descendantsCount || 0);
				if (low <= index && index <= nextLow)
				{
					i = addDomNodesHelp(childNodes[j], vChild, patches, i, low, nextLow, eventNode);
					if (!(patch = patches[i]) || (index = patch.index) > high)
					{
						return i;
					}
				}
				low = nextLow;
			}
			return i;
```

In the call of `addDomNodesHelp` just before the crash we can see that `vNode` corresponds to a `div`, let's say a `div` with class `my-div`, but `domNode` corresponds to another `div` with class `backdrop` which, I guess, has been added by the dialog polyfill.
Because of that, `vChildren` is a non empty array, but `childNodes` is an empty array, and in the next call of `addDomNodesHelp` `childNodes[j]` is `undefined`.

It appears that I have this error because of the `.backdrop` `div` added by polyfill.

Did you already encounter this error? How can we avoid it? Is it because my `dialog` is called inside a `div` instead of the `body` element? 
But in this case, how can we add the `dialog` as a direct child of the `body` element? Because even if I embed the Elm application directly in the `body` tag, the root view must return a `Html Msg` which is what the `dialog` is already returning. That means that I can't add the rest of my view after the `dialog` without putting the `dialog` and the rest of my view inside a `div` (which takes a `List (Html Msg)`

## Edit

I know it's not directly a problem with elm-mdl, but it is advised in the `Dialog` documentation to use the google dialog polyfill. So I'm wondering if I'm using it correctly. And if I'm using it correctly, then I guess there should be a workaround explained in the documentation or a fix to allow the use of this polyfill without making Elm crash

**Comment by FabienHenon** [comment]
I could fix my issue by encapsulating the call to the `dialog` in a `Html.Keyed` div `node`.

Maybe it would be interesting to add this keyed node in the `dialog` code in the library?
Do you want me to write a PR?

**Comment by aforemny** [comment]
Hi!
That sounds good. I had trouble reproducing the issue though. Could you check out [this repo](https://github.com/aforemny/elm-mdl-issue-284) and tell me what's wrong. page.html runs fine in Chrome and Firefox on Linux for me.

**Comment by aforemny** [comment]
I will close here because there was no response. @FabienHenon, if you are getting back to this, please re-open. The fix you've suggested seems fine for me, but I would like to understand the issue first, ie. reproduce it.

**Comment by FabienHenon** [comment]
Hi,

Sorry for the late answer, I didn't see your message.

I don't have enough time to check your code by myself but from what I see, here are a few differences you have with my own code:
- I was not incuding `dialog-polyfill` from the CDN with feature `focusin`
- I was not running `Elm` in fullscreen mode
- I rendered the `dialog` before everything

If you still can't reproduce the issue, maybe it was an issue with Firefox and they fixed it, or maybe the issue is not present on linux (I'm using a Mac), or maybe your version of the polyfill is different and they fixed the issue (I'm using version 0.4.7 from npm)
